File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03017/p03017_num8_parsed.py
def can_achieve_objective(N, A, B, C, D, S):
    # Check if there is an obstacle between A and C
    if "#" in S[A-1:C]:
        return "No"
    
    # Check if there is an obstacle between B and D
    if "#" in S[B-1:D]:
        return "No"
    
    # Check if there is a rock exactly in D+1 and all squares after D+1 are empty
    if D < N and S[D] == "#" and "#" in S[D+1:]:
        return "No"
    
    # Check if there is a rock exactly in C+1 and all squares between A and C are empty
    if A <= C-2 and S[C-2] == "#" and "#" in S[A-1:C-2]:
        return "No"
    
    return "Yes"

# Read input
N, A, B, C, D = map(int, input().split())
S = input().strip()

# Call the function and print the result
result = can_achieve_objective(N, A, B, C, D, S)
print(result)

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03017/p03017_num8_parsed.cpp
#include <iostream>
#include <string>

using namespace std;

bool is_possible(int N, int A, int B, int C, int D, string& S) {
    // Check if there is a rock between A and B
    for (int i = A; i < B; i++) {
        if (S[i - 1] == '#') {
            return false;
        }
    }
    // Check if there is a rock between C and D
    for (int i = C; i < D; i++) {
        if (S[i - 1] == '#') {
            return false;
        }
    }
    // Check if there is a rock between B and C
    for (int i = B; i < C; i++) {
        if (S[i - 1] == '#') {
            return false;
        }
    }
    // Check if A-1 is reachable from A
    if (A + 1 < B && S[A + 1 - 1] == '.') {
        return true;
    }
    // Check if D+1 is reachable from D
    if (D + 1 <= N && S[D + 1 - 1] == '.') {
        return true;
    }
    // Check if A-2 is reachable from A
    if (A + 2 <= B && S[A + 2 - 1] == '.') {
        return true;
    }
    // Check if D+2 is reachable from D
    if (D + 2 <= N && S[D + 2 - 1] == '.') {
        return true;
    }
    return false;
}

int main() {
    int N, A, B, C, D;
    string S;
    cin >> N >> A >> B >> C >> D;
    cin >> S;

    if (is_possible(N, A, B, C, D, S)) {
        cout << "Yes" << endl;
    } else {
        cout << "No" << endl;
    }

    return 0;
}